### ISSUE TITLE
The Strategy of playing more than 32 audios at the same time needs to be changed

### ISSUE THREAD

**Issue by dumganhar** [cocos2d/cocos2d-x#17387]
Please refer to https://github.com/cocos2d/cocos2d-x/pull/17233

- cocos2d-x version: v3.14.1
- devices test on: Android devices
- developing environments
   - NDK version: r10e
   - Xcode version: N/A
   - VS version: N/A
   - browser type and version: N/A

Steps to Reproduce:

1. Just AudioEngine::play2d(...) as many audios as possible
2. The latest audio will not be played, there is an output log:
```
Fail to play xxx.mp3 cause by limited max instance of AudioEngine.
```

### HOW TO DO:

Basically, we need to modify the behavior of playing more than `MAX` instances (32).
Currently, we just ignore the latest audio added to AudioEngine, IMO, this strategy doesn't make sense. We probably could take a look at `SoundPool` implementation in Android, which removes the oldest audio added to `AudioEngine` and re-uses it for the latest one.

**Comment by piotrros** [comment]
+1 
I agree :D

**Comment by piotrros** [comment]
Any chance to add this "feature"? I'm really struggling with this issue A LOT.

**Comment by stevetranby** [comment]
Ideally one would mix the internal data buffer(s) if necessary (or always?) into a single or few channel(s) of actual playback. It's possible this is only practical on desktop and high-end mobile devices.
